bin/i18n/resources/dashboard/text_to_speech/sync_out.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+
+require 'json'
+
+require_relative '../../../../../dashboard/config/environment'
+require_relative '../../../../../dashboard/app/models/concerns/text_to_speech'
+require_relative '../../../i18n_script_utils'
+require_relative '../../../utils/sync_out_base'
+require_relative '../text_to_speech'
+
+module I18n
+  module Resources
+    module Dashboard
+      module TextToSpeech
+        class SyncOut < I18n::Utils::SyncOutBase
+          TTS_LOCALES = (::TextToSpeech::VOICES.keys - %i[en-US]).freeze
+          METRIC_CONTEXT = 'update_level_i18n'.freeze
+
+          def perform
+            progress_bar.start
+
+            progress_bar.total = tts_units.size
+            tts_units.find_each do |unit|
+              next unless unit.text_to_speech_enabled?
+
+              unit.levels.merge(Blockly.all).find_each do |level|
+                I18nScriptUtils.process_in_threads(TTS_LOCALES) do |locale|
+                  upload_tts_short_instructions_l10n(level, locale)
```

Comment:
"Internationalization (i18n) is the process of designing and developing software or products that can be adapted to different languages and cultures, while localization (l10n) is the process of adapting a product or content for a specific locale or market."

In this context we "localize" the content for one specific locale, and internationalization more like a global process.

I'll change it to internationalization if you say.

Reply:
I don't think those definitions quite reflect the way we use those terms internally; we consistently use internationalization to refer to anything related to supporting the user's ability to change the language in which the content is presented, which is what this code is meant to support. I think we should standardize around `i18n` in this part of the feature, and save `l10n` for scenarios in which we are authoring wholly different content for different regions.

Reply:
I thought since we upload `"localized"` strings, the name `upload_tts_short_instructions_l10n` (upload TTS short_insturctions `localization`) makes more sense in this context

Reply:
As discussed offline, that's a great point that we do indeed use the term "localization" in other parts of the codebase. I think we're not actually that consistent here, so I'll leave it up the folks who work in this area more to comment on whether we actually have any preferences here!

Reply:
Couple thoughts:
1. Do we need to suffix the function name with anything, or would `upload_tts_short_instructions` suffice?
2. Can we avoid having to decide between i18n and l10n by using "translation" when it makes sense? For example, `upload_translated_tts_short_instructions`.

Reply:
1. It adds more context about what the method does, and the right method name is the best documentation.
2. I'm afraid this will make it more confusing, because instead of two options we'll have three to choose

Reply:
Sticking to Rails documentation:
- i18n means to abstract all strings and other locale-specific bits (such as date or currency formats) out of your application.
- l10n means to provide translations and localized formats for these bits.
In this case, the method in discussion uploads translations. It does not localize nor internationalize the content. Being strict, the i18n pipeline just sets the elements required to localize content. However, if we want to be consistent with similar parts of our code base, when other translated resources are uploaded to an S3 bucket, the term, localized is used.
https://github.com/code-dot-org/code-dot-org/blame/staging/bin/animation_assets/manifest_builder.rb#:~:text=205-,def,),-last%20month